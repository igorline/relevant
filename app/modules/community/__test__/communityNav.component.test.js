import React from 'react';
import renderer from 'react-test-renderer';
import { Community } from 'modules/community/communityNav.component';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';
// import * as communityActions from 'modules/community/community.actions';
// import { setCommunity } from 'modules/auth/auth.actions';

jest.mock('modules/community/hooks', () => ({ useUnread: () => 5 }));

jest.mock('react-redux', () => {
  const { auth, community } = require('app/mockdata');
  const state = {
    auth: auth.auth,
    community: community.community,
    navigation: { screenSize: 0 }
  };

  return {
    useDispatch: () => () => {},
    useSelector: fn => fn(state),
    connect: el => el
  };
});

test('Snapshot NavProfile', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Community />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
