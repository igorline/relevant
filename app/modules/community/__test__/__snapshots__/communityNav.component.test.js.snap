// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot NavProfile 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: hsl(0,0%,0%);
  color: #0000ff;
}

.c0:hover > div > div:first-child {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-text-decoration-color: hsl(0,0%,0%);
  text-decoration-color: hsl(0,0%,0%);
}

.c0:hover {
  background: hsl(0,0%,100%);
}

.c2 {
  color: #0000ff;
}

.c1 {
  color: #0000ff;
  color: hsl(0,0%,50%);
}

.c1:hover * {
  color: hsl(0,0%,0%);
}

.c1:hover * {
  fill: hsl(0,0%,0%);
}

.c1.active * {
  color: hsl(0,0%,0%);
}

.c1.active * {
  fill: hsl(0,0%,0%);
}

.c1:hover * {
  color: hsl(0,0%,0%);
}

<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0px",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "display": "flex",
      "flexBasis": "0px",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0px",
      "position": "relative",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "borderBottomColor": "rgba(222,222,222,1.00)",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "1px",
        "borderLeftStyle": "solid",
        "borderRightStyle": "solid",
        "borderTopStyle": "solid",
        "display": "flex",
        "position": "relative",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,1.00)",
          "display": "flex",
          "marginRight": "1px",
          "position": "relative",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "display": "flex",
            "marginTop": "1.5rem",
            "position": "relative",
          }
        }
      />
      <a
        className="c0 ulink-mock"
        flex={1}
        href="/relevant/new"
        hu={0}
        inline={0}
        onClick={[Function]}
        styles={
          Array [
            "display:flex;align-items:center;color:",
            "hsl(0, 0%, 0%)",
            ";&:hover > div > div:first-child{text-decoration:underline;text-decoration-color:",
            "hsl(0, 0%, 0%)",
            ";}&:hover{background:",
            "hsl(0, 0%, 100%)",
            ";}",
          ]
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxDirection": "normal",
              "WebkitBoxFlex": 1,
              "WebkitBoxOrient": "horizontal",
              "WebkitBoxPack": "justify",
              "WebkitFlexBasis": "0px",
              "WebkitFlexDirection": "row",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitJustifyContent": "space-between",
              "alignItems": "center",
              "display": "flex",
              "flexBasis": "0px",
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "space-between",
              "msFlexAlign": "center",
              "msFlexDirection": "row",
              "msFlexNegative": 1,
              "msFlexPack": "justify",
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0px",
              "position": "relative",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "WebkitAlignItems": "center",
                "WebkitBoxAlign": "center",
                "WebkitBoxDirection": "normal",
                "WebkitBoxOrient": "horizontal",
                "WebkitFlexDirection": "row",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "msFlexAlign": "center",
                "msFlexDirection": "row",
                "paddingBottom": "0.5rem",
                "paddingLeft": "1.5rem",
                "paddingRight": "1.5rem",
                "paddingTop": "0.5rem",
                "position": "relative",
              }
            }
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "WebkitFlexBasis": "auto",
                  "backgroundColor": "rgba(128,128,128,1.00)",
                  "display": "flex",
                  "flexBasis": "auto",
                  "height": "2rem",
                  "marginRight": "0.75rem",
                  "msFlexPreferredSize": "auto",
                  "overflowX": "hidden",
                  "overflowY": "hidden",
                  "position": "relative",
                  "width": "2rem",
                  "zIndex": 0,
                }
              }
            >
              <div
                className="css-view-1dbjc4n"
                style={
                  Object {
                    "backgroundColor": "rgba(0,0,0,0.00)",
                    "backgroundImage": "url(\\"https://relevant-image.s3.amazonaws.com/9fgwyrd05_rL.jpg\\")",
                    "backgroundPosition": "center",
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "cover",
                    "bottom": "0px",
                    "height": "100%",
                    "left": "0px",
                    "position": "absolute",
                    "right": "0px",
                    "top": "0px",
                    "width": "100%",
                    "zIndex": -1,
                  }
                }
              />
              <img
                alt=""
                className="css-accessibilityImage-9pa8cd"
                draggable={false}
                src="https://relevant-image.s3.amazonaws.com/9fgwyrd05_rL.jpg"
              />
            </div>
            <div
              className="css-text-901oao"
              dir="auto"
              style={
                Object {
                  "WebkitTextDecorationColor": "rgba(0,0,0,1.00)",
                  "color": "rgba(0,0,0,1.00)",
                  "fontFamily": "HelveticaNeue",
                  "fontSize": "0.875rem",
                  "lineHeight": "0.875rem",
                  "textDecorationColor": "rgba(0,0,0,1.00)",
                }
              }
            >
              Relevant
            </div>
          </div>
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "WebkitAlignItems": "center",
                "WebkitBoxAlign": "center",
                "WebkitBoxPack": "center",
                "WebkitJustifyContent": "center",
                "alignItems": "center",
                "backgroundColor": "rgba(0,0,255,1.00)",
                "borderBottomLeftRadius": "0.4375rem",
                "borderBottomRightRadius": "0.4375rem",
                "borderTopLeftRadius": "0.4375rem",
                "borderTopRightRadius": "0.4375rem",
                "display": "flex",
                "height": "0.875rem",
                "justifyContent": "center",
                "marginRight": "1.5rem",
                "minWidth": "0.875rem",
                "msFlexAlign": "center",
                "msFlexPack": "center",
                "paddingBottom": "0rem",
                "paddingLeft": "0.25rem",
                "paddingRight": "0.25rem",
                "paddingTop": "0rem",
                "position": "relative",
              }
            }
          >
            <div
              className="css-text-901oao"
              dir="auto"
              style={
                Object {
                  "WebkitTextDecorationColor": "rgba(255,255,255,1.00)",
                  "color": "rgba(255,255,255,1.00)",
                  "display": "flex",
                  "fontFamily": "HelveticaNeue-CondensedBold",
                  "fontSize": "0.625rem",
                  "lineHeight": "0.875rem",
                  "position": "relative",
                  "textDecorationColor": "rgba(255,255,255,1.00)",
                }
              }
            >
              5
            </div>
          </div>
        </div>
      </a>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "borderBottomColor": "rgba(222,222,222,1.00)",
            "borderBottomStyle": "solid",
            "borderBottomWidth": "1px",
            "borderLeftStyle": "solid",
            "borderRightStyle": "solid",
            "borderTopStyle": "solid",
            "display": "flex",
            "paddingBottom": "2rem",
            "paddingLeft": "1.5rem",
            "paddingRight": "1.5rem",
            "paddingTop": "0rem",
            "position": "relative",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "display": "flex",
              "marginBottom": "0rem",
              "marginLeft": "2.75rem",
              "marginRight": "0rem",
              "marginTop": "0.25rem",
              "position": "relative",
            }
          }
        >
          <a
            ac="hsl(0, 0%, 0%)"
            aria-current={null}
            className="c1 ulink-mock"
            color="hsl(0, 0%, 50%)"
            hc="hsl(0, 0%, 0%)"
            href="/relevant/new/technology"
            hu={0}
            inline={0}
            onClick={[Function]}
            styles=""
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "marginBottom": "0.25rem",
                  "marginTop": "0.25rem",
                }
              }
            >
              <div
                className="css-text-901oao"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(128,128,128,1.00)",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": "0.875rem",
                    "lineHeight": "1.2249999999999999rem",
                  }
                }
              >
                #
                technology
              </div>
            </div>
          </a>
          <a
            ac="hsl(0, 0%, 0%)"
            aria-current={null}
            className="c1 ulink-mock"
            color="hsl(0, 0%, 50%)"
            hc="hsl(0, 0%, 0%)"
            href="/relevant/new/society"
            hu={0}
            inline={0}
            onClick={[Function]}
            styles=""
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "marginBottom": "0.25rem",
                  "marginTop": "0.25rem",
                }
              }
            >
              <div
                className="css-text-901oao"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(128,128,128,1.00)",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": "0.875rem",
                    "lineHeight": "1.2249999999999999rem",
                  }
                }
              >
                #
                society
              </div>
            </div>
          </a>
          <a
            ac="hsl(0, 0%, 0%)"
            aria-current={null}
            className="c1 ulink-mock"
            color="hsl(0, 0%, 50%)"
            hc="hsl(0, 0%, 0%)"
            href="/relevant/new/culture"
            hu={0}
            inline={0}
            onClick={[Function]}
            styles=""
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "marginBottom": "0.25rem",
                  "marginTop": "0.25rem",
                }
              }
            >
              <div
                className="css-text-901oao"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(128,128,128,1.00)",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": "0.875rem",
                    "lineHeight": "1.2249999999999999rem",
                  }
                }
              >
                #
                culture
              </div>
            </div>
          </a>
          <a
            ac="hsl(0, 0%, 0%)"
            aria-current={null}
            className="c1 ulink-mock"
            color="hsl(0, 0%, 50%)"
            hc="hsl(0, 0%, 0%)"
            href="/relevant/new/politics"
            hu={0}
            inline={0}
            onClick={[Function]}
            styles=""
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "marginBottom": "0.25rem",
                  "marginTop": "0.25rem",
                }
              }
            >
              <div
                className="css-text-901oao"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(128,128,128,1.00)",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": "0.875rem",
                    "lineHeight": "1.2249999999999999rem",
                  }
                }
              >
                #
                politics
              </div>
            </div>
          </a>
          <a
            ac="hsl(0, 0%, 0%)"
            aria-current={null}
            className="c1 ulink-mock"
            color="hsl(0, 0%, 50%)"
            hc="hsl(0, 0%, 0%)"
            href="/relevant/new/altwoke"
            hu={0}
            inline={0}
            onClick={[Function]}
            styles=""
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "marginBottom": "0.25rem",
                  "marginTop": "0.25rem",
                }
              }
            >
              <div
                className="css-text-901oao"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(128,128,128,1.00)",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": "0.875rem",
                    "lineHeight": "1.2249999999999999rem",
                  }
                }
              >
                #
                altwoke
              </div>
            </div>
          </a>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "display": "flex",
              "marginTop": "1.5rem",
              "position": "relative",
            }
          }
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "fontFamily": "HelveticaNeue",
                "fontSize": "0.875rem",
                "lineHeight": "1.2249999999999999rem",
              }
            }
          >
            <span>
              A community for curating and discussing links related to critical analysis of #technology, #society and #culture.
            </span>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxDirection": "normal",
              "WebkitBoxOrient": "horizontal",
              "WebkitBoxPack": "justify",
              "WebkitFlexDirection": "row",
              "WebkitJustifyContent": "space-between",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": "1rem",
              "marginTop": "1.5rem",
              "msFlexDirection": "row",
              "msFlexPack": "justify",
              "position": "relative",
            }
          }
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "WebkitTextDecorationColor": "rgba(0,0,0,1.00)",
                "color": "rgba(0,0,0,1.00)",
                "fontFamily": "HelveticaNeue",
                "fontSize": "0.875rem",
                "lineHeight": "1.2249999999999999rem",
                "textDecorationColor": "rgba(0,0,0,1.00)",
              }
            }
          >
            60 Members
          </div>
          <a
            className="c2 ulink-mock"
            href="/"
            hu={0}
            inline={1}
            onClick={[Function]}
            styles=""
          >
            <div
              className="css-text-901oao"
              dir="auto"
              style={
                Object {
                  "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
                  "color": "rgba(0,0,255,1.00)",
                  "fontFamily": "HelveticaNeue",
                  "fontSize": "0.875rem",
                  "lineHeight": "1.2249999999999999rem",
                  "textDecorationColor": "rgba(0,0,255,1.00)",
                }
              }
            >
              See All
            </div>
          </a>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxDirection": "normal",
              "WebkitBoxLines": "multiple",
              "WebkitBoxOrient": "horizontal",
              "WebkitFlexDirection": "row",
              "WebkitFlexWrap": "wrap",
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
              "msFlexDirection": "row",
              "msFlexWrap": "wrap",
              "position": "relative",
            }
          }
        />
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "WebkitTextDecorationColor": "rgba(0,0,255,1.00)",
              "color": "rgba(0,0,255,1.00)",
              "fontFamily": "HelveticaNeue",
              "fontSize": "0.75rem",
              "lineHeight": "1.0499999999999998rem",
              "marginTop": "1rem",
              "textDecorationColor": "rgba(0,0,255,1.00)",
            }
          }
        >
          Settings
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "flex",
          "marginBottom": "1rem",
          "marginLeft": "0rem",
          "marginRight": "0rem",
          "marginTop": "1rem",
          "position": "relative",
        }
      }
    >
      <a
        className="c0 ulink-mock"
        flex={1}
        href="/crypto/new"
        hu={0}
        inline={0}
        onClick={[Function]}
        styles={
          Array [
            "display:flex;align-items:center;color:",
            "hsl(0, 0%, 0%)",
            ";&:hover > div > div:first-child{text-decoration:underline;text-decoration-color:",
            "hsl(0, 0%, 0%)",
            ";}&:hover{background:",
            "hsl(0, 0%, 100%)",
            ";}",
          ]
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitAlignItems": "center",
              "WebkitBoxAlign": "center",
              "WebkitBoxDirection": "normal",
              "WebkitBoxFlex": 1,
              "WebkitBoxOrient": "horizontal",
              "WebkitBoxPack": "justify",
              "WebkitFlexBasis": "0px",
              "WebkitFlexDirection": "row",
              "WebkitFlexGrow": 1,
              "WebkitFlexShrink": 1,
              "WebkitJustifyContent": "space-between",
              "alignItems": "center",
              "display": "flex",
              "flexBasis": "0px",
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "space-between",
              "msFlexAlign": "center",
              "msFlexDirection": "row",
              "msFlexNegative": 1,
              "msFlexPack": "justify",
              "msFlexPositive": 1,
              "msFlexPreferredSize": "0px",
              "position": "relative",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "WebkitAlignItems": "center",
                "WebkitBoxAlign": "center",
                "WebkitBoxDirection": "normal",
                "WebkitBoxOrient": "horizontal",
                "WebkitFlexDirection": "row",
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "msFlexAlign": "center",
                "msFlexDirection": "row",
                "paddingBottom": "0.5rem",
                "paddingLeft": "1.5rem",
                "paddingRight": "1.5rem",
                "paddingTop": "0.5rem",
                "position": "relative",
              }
            }
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "WebkitFlexBasis": "auto",
                  "backgroundColor": "rgba(128,128,128,1.00)",
                  "display": "flex",
                  "flexBasis": "auto",
                  "height": "2rem",
                  "marginRight": "0.75rem",
                  "msFlexPreferredSize": "auto",
                  "overflowX": "hidden",
                  "overflowY": "hidden",
                  "position": "relative",
                  "width": "2rem",
                  "zIndex": 0,
                }
              }
            >
              <div
                className="css-view-1dbjc4n"
                style={
                  Object {
                    "backgroundColor": "rgba(0,0,0,0.00)",
                    "backgroundImage": "url(\\"default_community.png\\")",
                    "backgroundPosition": "center",
                    "backgroundRepeat": "no-repeat",
                    "backgroundSize": "cover",
                    "bottom": "0px",
                    "height": "100%",
                    "left": "0px",
                    "position": "absolute",
                    "right": "0px",
                    "top": "0px",
                    "width": "100%",
                    "zIndex": -1,
                  }
                }
              />
              <img
                alt=""
                className="css-accessibilityImage-9pa8cd"
                draggable={false}
                src="default_community.png"
              />
            </div>
            <div
              className="css-text-901oao"
              dir="auto"
              style={
                Object {
                  "WebkitTextDecorationColor": "rgba(0,0,0,1.00)",
                  "color": "rgba(0,0,0,1.00)",
                  "fontFamily": "HelveticaNeue",
                  "fontSize": "0.875rem",
                  "lineHeight": "0.875rem",
                  "textDecorationColor": "rgba(0,0,0,1.00)",
                }
              }
            >
              Crypto
            </div>
          </div>
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "WebkitAlignItems": "center",
                "WebkitBoxAlign": "center",
                "WebkitBoxPack": "center",
                "WebkitJustifyContent": "center",
                "alignItems": "center",
                "backgroundColor": "rgba(0,0,255,1.00)",
                "borderBottomLeftRadius": "0.4375rem",
                "borderBottomRightRadius": "0.4375rem",
                "borderTopLeftRadius": "0.4375rem",
                "borderTopRightRadius": "0.4375rem",
                "display": "flex",
                "height": "0.875rem",
                "justifyContent": "center",
                "marginRight": "1.5rem",
                "minWidth": "0.875rem",
                "msFlexAlign": "center",
                "msFlexPack": "center",
                "paddingBottom": "0rem",
                "paddingLeft": "0.25rem",
                "paddingRight": "0.25rem",
                "paddingTop": "0rem",
                "position": "relative",
              }
            }
          >
            <div
              className="css-text-901oao"
              dir="auto"
              style={
                Object {
                  "WebkitTextDecorationColor": "rgba(255,255,255,1.00)",
                  "color": "rgba(255,255,255,1.00)",
                  "display": "flex",
                  "fontFamily": "HelveticaNeue-CondensedBold",
                  "fontSize": "0.625rem",
                  "lineHeight": "0.875rem",
                  "position": "relative",
                  "textDecorationColor": "rgba(255,255,255,1.00)",
                }
              }
            >
              5
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className="css-text-901oao"
    dir="auto"
    style={
      Object {
        "color": "rgba(0,0,0,1.00)",
        "fontFamily": "HelveticaNeue",
        "fontSize": "0.875rem",
        "lineHeight": "1.2249999999999999rem",
        "marginBottom": "1.5rem",
        "marginLeft": "1.5rem",
        "marginRight": "1.5rem",
        "marginTop": "1.5rem",
      }
    }
  >
    We'll be adding more communities soon!
    


  </div>
</div>
`;
